src/components/tables/cells/refreshTokens/RevokeToken.tsx

```diff
@@ -0,0 +1,92 @@
+import { LoadingButton } from '@mui/lab';
+import { Stack, TableCell, Tooltip, useTheme } from '@mui/material';
+import { PostgrestError } from '@supabase/postgrest-js';
+import { INVALID_TOKEN_INTERVAL, updateRefreshTokenValidity } from 'api/tokens';
+import Error from 'components/shared/Error';
+import { sample_blue } from 'context/Theme';
+import { useZustandStore } from 'context/Zustand/provider';
+import { WarningCircle } from 'iconoir-react';
+import { useState } from 'react';
+import { FormattedMessage } from 'react-intl';
+import {
+    SelectableTableStore,
+    selectableTableStoreSelectors,
+} from 'stores/Tables/Store';
+import { SelectTableStoreNames } from 'stores/names';
+
+interface Props {
+    id: string;
+}
+
+function RevokeTokenButton({ id }: Props) {
+    const theme = useTheme();
+
+    const hydrate = useZustandStore<
+        SelectableTableStore,
+        SelectableTableStore['hydrate']
+    >(
+        SelectTableStoreNames.REFRESH_TOKENS,
+        selectableTableStoreSelectors.query.hydrate
+    );
+
+    const [saving, setSaving] = useState(false);
+    const [error, setError] = useState<PostgrestError | null>(null);
+
+    const revokeToken = async (event: React.MouseEvent<HTMLElement>) => {
+        event.preventDefault();
+
+        setSaving(true);
+        setError(null);
+
+        const response = await updateRefreshTokenValidity(
+            id,
+            INVALID_TOKEN_INTERVAL
+        );
+
+        if (response.error) {
+            setError(response.error);
+            setSaving(false);
+
+            return;
+        }
+
+        hydrate();
```

Comment:
This should be `awaited`

Reply:
This select table store action is improperly typed -- returning `void` instead of a promise -- therefore it cannot be awaited. Changing the type annotation has ripple effects across all select table stores.

Reply:
you could just await this and then mark it as an error expected

Reply:
If you're okay with this getting ignored we're fine - so it is up to you